src/test/java/io/anserini/integration/IndexerTest.java

```diff
@@ -149,6 +159,70 @@ public void testReadingPostings() throws Exception {
     reader.close();
   }
 
+  // This test case iterates through all documents in the index and prints out the document vector:
+  // For each term, we print out the term frequency.
+  @Test
+  public void testIterateThroughDocumentVector() throws Exception {
+    Directory dir = FSDirectory.open(tempDir1);
+    IndexReader reader = DirectoryReader.open(dir);
+
+    int numDocs = reader.numDocs();
+    // Iterate through the document vectors
+    for (int i=0; i<numDocs; i++) {
+      System.out.println(reader.document(i));
+      Terms terms = reader.getTermVector(i, "text");
+      TermsEnum te = terms.iterator();
+
+      // For this document, iterate through the terms.
+      Term term;
+      while (te.next() != null) {
+        term = new Term("text", te.term());
+        long tf = te.totalTermFreq();
+        // Print out the term and its term frequency
+        System.out.println(term.bytes().utf8ToString() + " " + tf);
+      }
+    }
+  }
+
+  // This test case iterates through all documents in the index and prints out the document vector:
+  // For each term, we print out the term frequency and the BM25 weight.
+  @Test
+  public void testIterateThroughDocumentVectorComputeBM25() throws Exception {
+    Directory dir = FSDirectory.open(tempDir1);
+    IndexReader reader = DirectoryReader.open(dir);
+    IndexSearcher searcher = new IndexSearcher(reader);
+    searcher.setSimilarity(new BM25Similarity());
+
+    int numDocs = reader.numDocs();
+    // Iterate through the document vectors
+    for (int i=0; i<numDocs; i++) {
+      String docid = reader.document(i).getField("docid").stringValue();
+      System.out.println(reader.document(i));
+      System.out.println(i+ ": " + docid);
+      Terms terms = reader.getTermVector(i, "text");
+      TermsEnum te = terms.iterator();
+
+      // For this document, iterate through the terms.
+      while (te.next() != null) {
+        String term = new Term("text", te.term()).bytes().utf8ToString();
+        String stemmed = AnalyzerUtils.tokenize(new EnglishAnalyzer(), term).get(0);
```

Comment:
This is redundant. The term stored in the Lucene index should already be tokenized and stemmed.